src/test/java/net/snowflake/client/jdbc/PreparedStatement1IT.java

```diff
@@ -94,6 +80,46 @@ public void testGetParameterMetaData() throws SQLException {
     }
   }
 
+  /**
+   * Test to ensure it's possible to upload Time values via stage array binding and get proper
+   * values back (SNOW-194437)
+   *
+   * <p>Ignored on GitHub Action because CLIENT_STAGE_ARRAY_BINDING_THRESHOLD parameter is not
+   * available to customers so cannot be set when running on Github Action
+   *
+   * @throws SQLException
+   */
+  @Test
+  @ConditionalIgnoreRule.ConditionalIgnore(condition = RunningOnGithubAction.class)
+  public void testInsertStageArrayBindWithTime() throws SQLException {
+    try (Connection connection = init()) {
+      Statement statement = connection.createStatement();
+      statement.execute("alter session set CLIENT_STAGE_ARRAY_BINDING_THRESHOLD=2");
+      statement.execute("create or replace table testStageBindTime (c1 time, c2 date)");
+      PreparedStatement prepSt =
+          connection.prepareStatement("insert into testStageBindTime values (?, ?), (?, ?)");
+      prepSt.setTime(1, new Time(1));
+      prepSt.setDate(2, new Date(1));
+      prepSt.addBatch();
+      prepSt.setTime(1, new Time(100));
+      prepSt.setDate(2, new Date(100));
```

Comment:
can you add the tests for:
- negative epoch time
- zero
- very large value and very small, e.g., max int, negative max int
to capture the behaviors? I guess the negative value for Time would not work, though, we should know the behaviors.

Reply:
Added tests for all of these, although negative values now throw exceptions.